lib/api/dsl/index.js

```diff
@@ -11,90 +32,90 @@ const
  * one, while plugins can instanciate DSL engines
  * each one with its own hash key.
  *
- * @constructor
+ * @class RealtimeEngine
  * @param {string} [hashKey]
  */
-function RealtimeEngine (hashKey) {
-  this.hashKey = hashKey || crypto.randomBytes(32);
-  this.transformer = new Transformer();
-  this.storage = new Storage(this.hashKey);
-  this.matcher = new Matcher(this.storage);
+class RealtimeEngine {
+  constructor(hashKey) {
+    this.hashKey = hashKey || crypto.randomBytes(32);
+    this.transformer = new Transformer();
+    this.storage = new Storage(this.hashKey);
+    this.matcher = new Matcher(this.storage);
+  }
 
-  return this;
-}
+  /**
+   * Checks if the provided filters are valid
+   *
+   * @param {object} filters
+   * @return {Promise<Object>}
+   */
+  validate(filters) {
+    return this.transformer.check(filters);
+  }
 
-/**
- * Checks if the provided filters are valid
- *
- * @param {object} filters
- * @return {Promise<Object>}
- */
-RealtimeEngine.prototype.validate = function validate (filters) {
-  return this.transformer.check(filters);
-};
+  /**
+   * Subscribes a filter to the real-time engine
+   *
+   * @param {string} index
+   * @param {string} collection
+   * @param {object} filters
+   * @return {Promise}
+   */
+  register(index, collection, filters) {
+    return this.transformer.normalize(filters)
+      .then(normalized => this.storage.store(index, collection, normalized));
+  }
 
-/**
- * Subscribes a filter to the real-time engine
- *
- * @param {string} index
- * @param {string} collection
- * @param {object} filters
- * @return {Promise}
- */
-RealtimeEngine.prototype.register = function register (index, collection, filters) {
-  return this.transformer.normalize(filters)
-    .then(normalized => this.storage.store(index, collection, normalized));
-};
+  /**
+   * Check if there are filters registered on an index-collection pair
+   *
+   * @param index
+   * @param collection
+   * @returns {boolean}
+   */
+  exists(index, collection) {
+    return this.storage.filtersIndex[index] !== undefined && this.storage.filtersIndex[index][collection] !== undefined;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 120 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)